backend/app/services/user_info.rb

```diff
@@ -5,6 +5,23 @@
 
 class UserInfo
 
+  def self.mock_users
+    [
+      { user_id: '00000000-0000-0000-0000-000000000000', role: 'admin', first_name: 'Admin',
+        last_name: 'Uno', name: 'Admin Uno', full_name: 'Admin Uno' },
+      { user_id: '00000000-0000-0000-0000-000000000001', role: 'researcher', name: 'Researcher Uno',
+        full_name: 'Researcher Uno', first_name: 'Researcher', last_name: 'Uno' },
+      { user_id: '00000000-0000-0000-0000-000000000002', role: 'user', name: 'User Uno',
+        full_name: 'User Uno', first_name: 'User', last_name: 'Uno' },
+      { user_id: '00000000-0000-0000-0000-000000000003', role: 'user', name: 'User Dos',
+        full_name: 'User Dos', first_name: 'User', last_name: 'Dos' },
+      { user_id: '00000000-0000-0000-0000-000000000004', role: 'user', name: 'User Tres',
+        full_name: 'User Tres', first_name: 'User', last_name: 'Tres' },
+      { user_id: '00000000-0000-0000-0000-000000000005', role: 'user', name: 'User Cuatro',
+        full_name: 'User Cuatro', first_name: 'User', last_name: 'Cuatro' }
+    ].freeze
+  end
```

Comment:
since these are static they should be stored in a yaml file in config/ then assigned to a constant, ie:
```ruby
class UserInfo
    MOCK_USERS = YAML.load_file(Rails.root.join("config/mock-users.yaml"))
```
Also, why not make the list be a hash with uuid key so it's easier to search.   You could transform an array into hash via:
YAML.load_file(Rails.root.join("config/mock-users.yaml"))
  .reduce({}) { |h, u| h[u[:uuid]] = u; h }